scripts/generate-sponsorship-csv.ts

```diff
@@ -33,7 +52,9 @@ function countStars(stars: string): number {
   return (stars.match(/⭐/g) || []).length
 }
 
-function calculateSponsorship(lastFourWeeks: WeeklyData[]): {
+function calculateSponsorship(
+  lastFourWeeks: { endDate: string; data: WeeklyData }[],
```

Comment:
we don't ever use the word data (it's too generic, same as naming a variable `variable`)

We also only consider weeks by the startDate (it's inconvenient in this case, but it's best to be consistent everywhere)

the first `startDate` will sometimes be from the previous month. You might find you need to calculate endDate as well

```suggestion
  recentContributions: { startDate: string; endDate: string; contributionsByUser: ContributionsByUser }[],
```